src/vmm/src/builder.rs

```diff
@@ -508,30 +506,15 @@ pub fn setup_serial_device(
 #[cfg(target_arch = "x86_64")]
 fn attach_legacy_devices(
     vm: &Vm,
-    pio_device_manager: &mut PortIODeviceManager,
-) -> std::result::Result<(), StartMicrovmError> {
-    pio_device_manager
-        .register_devices()
-        .map_err(Error::LegacyIOBus)
-        .map_err(StartMicrovmError::Internal)?;
-
-    macro_rules! register_irqfd_evt {
-        ($evt: ident, $index: expr) => {{
-            vm.fd()
-                .register_irqfd(&pio_device_manager.$evt, $index)
-                .map_err(|e| {
-                    Error::LegacyIOBus(device_manager::legacy::Error::EventFd(
-                        io::Error::from_raw_os_error(e.errno()),
-                    ))
-                })
-                .map_err(StartMicrovmError::Internal)?;
-        }};
-    }
-
-    register_irqfd_evt!(com_evt_1_3, 4);
-    register_irqfd_evt!(com_evt_2_4, 3);
-    register_irqfd_evt!(kbd_evt, 1);
-    Ok(())
+    serial: Arc<Mutex<devices::legacy::Serial>>,
```

Comment:
The naming is not consistent with impl: creates a PIO device manager and also attaches devices.

Reply:
done